src/main/scala/com/github/unchama/seichiassist/subsystems/gacha/application/actions/LotteryOfGachaItems.scala

```diff
@@ -0,0 +1,16 @@
+package com.github.unchama.seichiassist.subsystems.gacha.application.actions
+
+import cats.effect.concurrent.Ref
+import com.github.unchama.seichiassist.subsystems.gacha.domain.GachaPrize
+
+/**
+ * ガチャアイテムの抽選を行う作用
+ */
+trait LotteryOfGachaItems[F[_], ItemStack] {
+
+  def runLottery(
+    amount: Int,
+    gachaPrizesListRepository: Ref[F, Vector[GachaPrize[ItemStack]]]
```

Comment:
```Scala
type GlobalGachaPrizeList[F[_], ItemStack] = Ref[F, Vector[GachaPrize[ItemStack]]]
```

のように type alias を張って、これを引数に指定した方がいいかも

```suggestion
    gachaPrizesListRepository: GlobalGachaPrizeList[F, ItemStack]
```

のような型に変えていいかも